### PR TITLE
refactor(Identity): refactor Identity into rxAuth

### PR DESCRIPTION
**Pull request by cathysiller**
JIRA: https://jira.rax.io/browse/FRMW-1150

### LGTMs
- [x] Dev LGTM
- [x] Dev+Vidyo LGTM

# rxAuth

### Before
rxauth-before

### After
rxauth-after

### CODE REVIEW

File: src/utilities/rxAuth/scripts/rxAuth.js [hunk 1]
```diff
@@ -3,16 +3,21 @@ angular.module('encore.ui.utilities')
  * @ngdoc service
  * @name utilities.service:rxAuth
  * @description
+ * <h3>rxAuth</h3>
```
Comment (CITguy) [inline src/utilities/rxAuth/scripts/rxAuth.js]: An `<h3>` is probably not necessary. This documentation is only for `rxAuth`.

File: src/utilities/rxAuth/scripts/rxAuth.js [hunk 2]
```diff
@@ -3,16 +3,21 @@ angular.module('encore.ui.utilities')
...
  * Service which provides an entire solution for authenticating, user session management
  * and permissions in the UI.  The rxAuth service is a wrapper for the Identity, Session and
  * Permission services.  These services were broken into smaller components to facilitate
  * customization and re-use.
+ * 
+ * <h3>Identity</h3>
+ * This is a component designed to aid interaction with Rackspace's Identity API.
```
Comment (CITguy) [inline src/utilities/rxAuth/scripts/rxAuth.js]: This addition is unnecessary. The documentation is only for `rxAuth`.

File: src/utilities/rxAuth/scripts/rxAuth.js [hunk 3]
```diff
@@ -3,16 +3,21 @@ angular.module('encore.ui.utilities')
...
  *
  * @requires utilities.service:Identity
  * @requires utilities.service:Session
  * @requires utilities.service:Permission
  *
  * @example
+ * <h3>rxAuth</h3>
```
Comment (CITguy) [inline src/utilities/rxAuth/scripts/rxAuth.js]: Unnecessary. Please remove.

File: src/utilities/rxAuth/scripts/rxAuth.js [hunk 4]
```diff
@@ -24,11 +29,42 @@ angular.module('encore.ui.utilities')
  * rxAuth.getRoles() // Returns an array of roles for a user
  * rxAuth.hasRole(role) // Returns true/false if user has specified role
  * </pre>
+ * 
+ * <h3>Identity</h3>
+ * <pre>
+ * Identity.loginWithJSON(json); //Returns a promise
+ * Identity.login({username: '', password: '', successCallback, errorCallback}); // returns a promise
```
Comment (CITguy) [inline src/utilities/rxAuth/scripts/rxAuth.js]: This is already documented above on line 22-23. Please remove this newly added documentation.

File: src/utilities/rxAuth/scripts/rxAuth.spec.js [hunk 5]
```diff
@@ -177,3 +171,47 @@ describe('utilities:Auth (DEPRECATED)', function () {
...
+    beforeEach(function () {
+        module('encore.ui.utilities');
+
+        inject(function ($injector) {
+            $httpBackend = $injector.get('$httpBackend');
+            identity = $injector.get('Identity');
+        });
+    });
+
+    describe('service:Identity', function () {
```
Comment (CITguy) [inline src/utilities/rxAuth/scripts/rxAuth.spec.js]: Please add these tests to the `rxAuth` specs. These are currently only ensuring that the legacy `Identity` service is still working as expected, but we still need to test that it behaves the same way as `rxAuth`.

File: src/utilities/rxAuth/scripts/rxAuth.js [hunk 6]
```diff
@@ -3,16 +3,21 @@ angular.module('encore.ui.utilities')
...
+ * <h3>rxAuth</h3>
  * Service which provides an entire solution for authenticating, user session management
  * and permissions in the UI.  The rxAuth service is a wrapper for the Identity, Session and
  * Permission services.  These services were broken into smaller components to facilitate
  * customization and re-use.
+ * 
+ * <h3>Identity</h3>
+ * This is a component designed to aid interaction with Rackspace's Identity API.
  *
  * @requires utilities.service:Identity
```
Comment (CITguy) [inline src/utilities/rxAuth/scripts/rxAuth.js]: You can remove this `@requires` as well. We no longer depend on the `Identity` service. In fact, it's the other way around. 😄

File: src/utilities/rxAuth/scripts/rxAuth.js (hunk 2 above)
Comment (cathysiller) [inline src/utilities/rxAuth/scripts/rxAuth.js]: removed

File: src/utilities/rxAuth/scripts/rxAuth.js (hunk 6 above)
Comment (cathysiller) [inline src/utilities/rxAuth/scripts/rxAuth.js]: removed @requires

File: src/utilities/rxAuth/scripts/rxAuth.js (hunk 4 above)
Comment (cathysiller) [inline src/utilities/rxAuth/scripts/rxAuth.js]: removed

File: src/utilities/rxAuth/scripts/rxAuth.js (hunk 3 above)
Comment (cathysiller) [inline src/utilities/rxAuth/scripts/rxAuth.js]: removed

File: src/utilities/rxAuth/scripts/rxAuth.js (hunk 1 above)
Comment (cathysiller) [inline src/utilities/rxAuth/scripts/rxAuth.js]: removed

File: src/utilities/rxAuth/scripts/rxAuth.spec.js (hunk 5 above)
Comment (cathysiller) [inline src/utilities/rxAuth/scripts/rxAuth.spec.js]: updated tests

File: src/utilities/rxAuth/scripts/rxAuth.spec.js [hunk 7]
```diff
@@ -177,3 +177,47 @@ describe('utilities:Auth (DEPRECATED)', function () {
         });
     });
 });
+
+describe('utilities:Identity (DEPRECATED)', function () {
```
Comment (CITguy) [inline src/utilities/rxAuth/scripts/rxAuth.spec.js]: I'd put this describe block back at `src/utilities/Identity/scripts/Identity.spec.js`. It'll make it easier to remove later.

**Comment by halkon** [comment]
Vidyo LGTM

**Comment by lentusignavus** [comment]
DEV LGTM

**Pull request merged**